File: serve_annotate.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const ROOT = 'C:/Users/19853/WorkBuddy/2026-07-18-07-52-25/firefightAI';
const PORT = 5800;

const MIME = {
  '.html': 'text/html; charset=utf-8',
  '.js': 'application/javascript',
  '.css': 'text/css',
  '.png': 'image/png',
  '.jpg': 'image/jpeg',
  '.json': 'application/json',
};

http.createServer((req, res) => {
  let url = req.url === '/' ? '/src/web/annotate_flags.html' : req.url;
  let filePath = path.join(ROOT, url);
  
  fs.readFile(filePath, (err, data) => {
    if (err) {
      res.writeHead(404);
      res.end('Not found: ' + url);
      return;
    }
    let ext = path.extname(filePath);
    res.writeHead(200, { 'Content-Type': MIME[ext] || 'application/octet-stream' });
    res.end(data);
  });
}).listen(PORT, () => console.log(`Server: http://localhost:${PORT}`));

File: FirefightAI_完整项目包/serve_annotate.py
import http.server, socketserver, os
os.chdir('C:/Users/19853/WorkBuddy/2026-07-18-07-52-25/firefightAI')

class H(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        if self.path in ('/', '/index.html'):
            self.path = '/src/web/annotate_flags.html'
        return super().do_GET()

socketserver.TCPServer.allow_reuse_address = True
httpd = socketserver.TCPServer(('0.0.0.0', 5800), H)
print('SERVER_READY')
httpd.serve_forever()
